### DISCUSSION TITLE
`getFirstMediaUrl('default', 'thumb')` vs `getFirstMedia('default')->getUrl('thumb')`

### DISCUSSION THREAD

**Discussion by bilogic** [spatie/laravel-medialibrary#2564] [Ideas]
On v9,

1. `getFirstMedia('default')->getUrl('thumb')` creates and returns the thumbnail URL
2. `getFirstMediaUrl('default', 'thumb')` does not create the thumbnail, returns original URL

Is 2 by design? Feels like a bug to me as I assumed it to be equivalent to 1. Thank you.

**Comment by freekmurze** [comment]
Thanks for reporting this.

I would accept a PR that lets 2. return the equivalent of 1.
